common/browser.go

```diff
@@ -550,24 +550,32 @@ func (b *Browser) NewContext(opts goja.Value) (*BrowserContext, error) {
 	defer span.End()
 
 	if b.context != nil {
-		return nil, errors.New("existing browser context must be closed before creating a new one")
+		err := errors.New("existing browser context must be closed before creating a new one")
+		SpanRecordError(span, "browserContext already exists", err)
+		return nil, err
 	}
 
 	action := target.CreateBrowserContext().WithDisposeOnDetach(true)
 	browserContextID, err := action.Do(cdp.WithExecutor(b.ctx, b.conn))
 	b.logger.Debugf("Browser:NewContext", "bctxid:%v", browserContextID)
 	if err != nil {
-		return nil, fmt.Errorf("creating browser context ID %s: %w", browserContextID, err)
+		err = fmt.Errorf("creating browser context ID %s: %w", browserContextID, err)
```

Comment:
```suggestion
		err := fmt.Errorf("creating browser context ID %s: %w", browserContextID, err)
```
Does this wrapped error have meaning outside of this block? If not, I'd suggest to shadow it here and in similar blocks below to avoid overriding the original value.

Reply:
👍 Shadowing it here can also prevent unintentional bugs if we never return here. The linter will catch that the shadowed variable is not used.

Reply:
Good point, fixed in 5b93c3a95c2761ff55e3f6e76a21b962744afee7